### ISSUE TITLE
Simpler use of KOKKOS_SEPARATE_LIBS for CMake

### ISSUE THREAD

**Issue by jjwilke** [kokkos/kokkos#2119]
I don't like the logic in the cmake-overhaul branch that I put in for handling single lib/versus separate libs. 

What probably makes more sense is to ALWAYS install the separate libs. Then, if we want a single libkokkos, we install it but have it depend on the individual libraries.

````
add_library(kokkos ...)
target_link_libraries(kokkos PUBLIC kokkoscore kokkoscontainers)
target_link_libraries(kokkos INTERFACE kokkosalgorithms)
````

This should work for both static and dynamic libraries. It also should still work even for people who do:
````
-L${KOKKOS} -lkokkos
````
explicitly.

**Comment by jjwilke** [comment]
@dalg24

**Comment by jjwilke** [comment]
Semi-ignoring this. CMake does not allow combining static libraries into a single library (even though it totally could).  The functionality is implemented as need in #2167

**Comment by masterleinad** [comment]
Addressed in #2447.

**Comment by jjwilke** [comment]
We have all agreed to just make kokkos an interface library for simplicity.
